### PR TITLE
Port CI to github actions

### PR DESCRIPTION
**Pull request by colinxfleming**
I rule and have completed some work on Case Manager that's ready for review!

Port our existing circle ci tests/checks into github actions, and depreciates a CI deploy in favor of a heroku deploy.

~I'm very interested in a second opinion on this, but it DOES run capybara/system tests cleanly. We lose parallel splitting but might be able to make that up with rails' native test suite parallelization (which we haven't rolled out yet but which I've been experimenting with in general).~ (this is still true but it's less of a problem, since rails has a built-in test parallelizer and I'm pretty close to implementing it in #2348 

~note that this also does not get the app into a deployable state, as we deploy to staging via circle and this doesn't fix that.~ (it does this now)

This splits the CI config into two different subjobs - one that runs static scanners like brakeman / ruby-audit / bundle-audit etc, and one that sets up the application and runs the test suite including system tests.

(In this current commit circle is still hooked up - when this gets approved I will unhook circleci and merge on a full green build.)

The reason we're doing this is because circleci made some under the hood changes that caused system tests to stop working almost entirely due to puma not starting. Troubleshooting this was a pretty frustrating experience, was above my level of expertise to fix, and knocked out our ability to follow our deploy process for two weeks or so. We had been meaning to take another pass at github actions for a little while, and now seemed like a good time.

This pull request makes the following changes:
* lots of github ci config
* zero out circleci config

no views
no issues

For reviewer:
* Adjust the title to explain what it does for the notification email to the listserv.
* Tag this PR:
  * `feature` if it contains a feature, fix, or similar. This is anything that contains a user-facing fix in some way, such as frontend changes, alterations to backend behavior, or bug fixes.
  * `dependencies` if it contains library upgrades or similar. This is anything that upgrades any dependency, such as a Gemfile update or npm package upgrade.
* If it contains neither, no need to tag this PR.

### CODE REVIEW

**Comment by colinxfleming** [comment]
okay, I think this gets us to a place where:
* builds fire
* tests run
* circle build will autodeploy to staging on merge (which sucks but does let us deploy again, until we can port over the deploy stuff to github actions too)

File: .github/workflows/test_dev_env.yml [hunk 1]
```diff
@@ -18,4 +19,85 @@ jobs:
...
+        uses: ruby/setup-ruby@v1 # automagically reads from .ruby-version
+        with:
+          bundler-cache: true
+
+      # npm/yarn setup
+      - name: Setup Node
+        uses: actions/setup-node@v1
+        with:
+          node-version: 14.2
+      - name: Find yarn cache location
```
Comment (DarthHater) [inline .github/workflows/test_dev_env.yml:55]: For caching go ahead and try this:

```
- uses: actions/cache@v2
  with:
    path: '**/node_modules'
    key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
```

File: .github/workflows/test_dev_env.yml [hunk 2]
```diff
@@ -18,4 +19,85 @@ jobs:
...
+        run: echo "::set-output name=dir::$(yarn cache dir)"
+      - name: Cache yarn
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.yarn-cache.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('**/yarn.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-yarn-
+      - name: yarn packages
+        run: |
```
Comment (DarthHater) [inline .github/workflows/test_dev_env.yml:66]: You can just do `run: yarn install`, the `|` is only if you need to do multiple commands

Comment (colinxfleming) [reply .github/workflows/test_dev_env.yml]: good flag, thank you!

File: .github/workflows/test_dev_env.yml (hunk 1 above)
Comment (colinxfleming) [reply .github/workflows/test_dev_env.yml]: https://github.com/DCAFEngineering/dcaf_case_management/pull/2361/commits/5080a9e0d02c43a9508e42050f216a7e42de439a addresses this, I think? I popped out the other two cache commands and am using your snippet here instead.

**Comment by colinxfleming** [comment]
Here's where this landed:
* Build CI completely swapped to github actions
* Deploy step removed and ported over to heroku - via a heroku-side config, we now listen for changes on main and push them
* Ditch CircleCI

This increases our build time by about 5m per run, but VASTLY simplifies our CI configuration and removes a service we use entirely. I'm also convinced we can get some of the time back with rails' test suite parallelization feature, which we haven't turned on yet (because of mongodb).

**Review by lomky — APPROVED** [review]
All glory to GitHub, may all our eggs live in this basket 😂😭

**Comment by colinxfleming** [comment]
whew

**Pull request merged**